gulp-core-build/src/tasks/GulpTask.ts

```diff
@@ -83,40 +110,98 @@ export abstract class GulpTask<TASK_CONFIG> implements IExecutable {
     }
   }
 
+  /**
+   * Overridable function which returns true if this task should be executed, or false if it should be skipped.
+   * @param buildConfig - the build configuration which should be used when determining if the task is enabled
+   * @returns true if the build is not redundant
+   */
   public isEnabled(buildConfig: IBuildConfig): boolean {
     return (!buildConfig || !buildConfig.isRedundantBuild);
   }
 
+  /**
+   * When the task is executed by the build system, this function is called once. Note that this function
+   * must either return a Promise, a Stream, or call the completeCallback() parameter.
```

Comment:
Nope, double check the `GulpTask.execute()` function. Calling `completeCallback()` is not necessary.